packages/eslint-plugin-ckeditor5-rules/lib/rules/ckeditor-plugin-flags.js

```diff
@@ -0,0 +1,680 @@
+			incorrectReturnValue: 'The method \'{{flag}}\' should return {{returnValue}} but returns {{actualValue}} instead.',
+			incorrectReturnLiteral: 'The method \'{{flag}}\' should return a literal value with the {{returnValue}} value.',
+			notPublicGetter: 'The method \'{{flag}}\' should be defined as a public getter method.',
+			singleReturnStatement: 'The method \'{{flag}}\' should have a single return statement.',
+			definedBeforePluginName: 'The method \'{{flag}}\' should be defined after the \'pluginName\' method.',
+			definedInProperOrder: 'The method \'{{flag}}\' should be defined directly after other flag method.',
+			definedAfterSpecificMethod: 'The method \'{{flag}}\' should be defined after the \'{{before}}\' method.'
+		}
+	},
+
+	create( context ) {
+		if ( !getRequiredFlagsConfig( context ).length && !getDisallowedFlagsConfig( context ).length ) {
+			return {};
+		}
+
+		return {
+			'ClassDeclaration:exit'( classDeclaration ) {
+				if ( isPluginClassDeclaration( classDeclaration ) ) {
+					analyzePluginClass( context, classDeclaration );
+				}
+			}
+		};
+	}
+};
+
+/**
+ * Returns the required flags from the rule options.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @returns {Array.<Object>}
+ */
+function getRequiredFlagsConfig( context ) {
+	return context.options[ 0 ]?.requiredFlags ?? [];
+}
+
+/**
+ * Returns the disallowed flags from the rule options.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @returns {Array.<String>}
+ */
+function getDisallowedFlagsConfig( context ) {
+	return context.options[ 0 ]?.disallowedFlags ?? [];
+}
+
+/**
+ * Analyzes the plugin class declaration and performs all necessary checks and reports errors if needed.
+ * What it does:
+ *
+ * 	* It checks if there are no disallowed flags.
+ * 	* It checks if all required flags are defined in the class declaration.
+ * 	* It checks if all required flags are defined after the `pluginName` method.
+ * 	* It checks if all required flags are defined in the proper order (after each other).
+ *
+ * While checking the order might be not super important, it's a good practice to keep the flags in the same order
+ * to avoid situations where the flags are scattered across the class. It unifies alignment of the methods with
+ * auto-fixer which enforces defining all flags after the `pluginName` method (mostly because it simplifies the fixer).
+ *
+ * @param {Object} context The ESLint rule context.
+ * @param {Object} classDeclaration The class declaration node to analyze.
+ */
+function analyzePluginClass( context, classDeclaration ) {
+	checkIfThereAreNoDisallowedFlags( context, classDeclaration );
+	checkIfAllFlagsAreProperlyDefined( context, classDeclaration );
+
+	const misalignedMethods = checkIfFlagsAreDefinedAfterPluginName( context, classDeclaration );
+
+	checkIfFlagsDefinedInProperOrder( context, classDeclaration, misalignedMethods );
```

Comment:
I'd not enforce this as it complicates this linter and probably won't be used to often.